front50-s3/src/main/java/com/netflix/spinnaker/front50/config/S3Config.java

```diff
@@ -83,52 +102,55 @@ public RestTemplate restTemplate() {
   }
 
   @Bean
-  public ObjectMapper amazonObjectMapper() {
-    return new AmazonObjectMapper();
+  public S3StorageService s3StorageService(AmazonS3 amazonS3) {
+    ObjectMapper awsObjectMapper = new ObjectMapper();
+    AmazonObjectMapperConfigurer.configure(awsObjectMapper);
+
+    return new S3StorageService(awsObjectMapper, amazonS3, bucket, rootFolder);
   }
 
   @Bean
-  public S3ApplicationDAO s3ApplicationDAO(ObjectMapper objectMapper, AmazonS3 amazonS3) {
-    return new S3ApplicationDAO(objectMapper, amazonS3, Schedulers.from(Executors.newFixedThreadPool(20)), 15000, bucket, rootFolder);
+  public ApplicationDAO applicationDAO(StorageService storageService) {
+    return new DefaultApplicationDAO(storageService, Schedulers.from(Executors.newFixedThreadPool(20)), 15000);
```

Comment:
How are the rates determined? Maybe these should be factored out as well. And possibly the number of threads. I imagine these are more behavioral (nature of the type and how it is used in spinnaker) than platform (s3 vs gcs). Maybe there can be a "small" and "large" config where one is for small projects that dont change frequently and one is for large projects that have a lot of data and changes more often. If that affects the tradeoffs made here.

Maybe not now, but ultimately it seems these might need to be tweakable for the extreme cases. Unless perhaps the rate is adaptable to the size of the domain or the update time (i.e. not constant). But that's not for this PR and depends on whether a suitable default for the extremes in use cases exists.